service/auth.ts

```diff
@@ -13,6 +13,30 @@ import {
 } from "proto/auth_pb";
 import client from "service/client";
 
+type CreateUserReq = {
+  email: string,
+  username: string,
+  password: string
+}
+
+type CreateUserRes = {
+  email: string,
+  username: string,
+  id: number
+}
+
+export type HttpError = {
+  status_code: number,
+  errors: [string],
```

Comment:
```suggestion
  errors: string[],
```
I think you probably want this instead. `[string]` literally means an array with one error message only which doesn't seem right given `errors` is plural.